lib/stats/DefaultStatsPresetPlugin.js

```diff
@@ -0,0 +1,227 @@
+	performance: NORMAL_ON,
+	hash: NORMAL_ON,
+	env: NORMAL_OFF,
+	version: NORMAL_ON,
+	timings: NORMAL_ON,
+	builtAt: NORMAL_ON,
+	assets: NORMAL_ON,
+	entrypoints: NORMAL_ON,
+	chunkGroups: OFF_FOR_TO_STRING,
+	chunks: OFF_FOR_TO_STRING,
+	chunkModules: OFF_FOR_TO_STRING,
+	chunkRootModules: ({ all, chunkModules }, { forToString }) => {
+		if (all === false) return false;
+		if (all === true) return true;
+		if (forToString && chunkModules) return false;
+		return true;
+	},
+	chunkOrigins: OFF_FOR_TO_STRING,
+	modules: NORMAL_ON,
+	nestedModules: OFF_FOR_TO_STRING,
+	orphanModules: NORMAL_OFF,
+	moduleAssets: OFF_FOR_TO_STRING,
+	depth: OFF_FOR_TO_STRING,
+	cached: NORMAL_ON,
+	runtime: OFF_FOR_TO_STRING,
+	cachedAssets: NORMAL_ON,
+	reasons: OFF_FOR_TO_STRING,
+	usedExports: OFF_FOR_TO_STRING,
+	providedExports: OFF_FOR_TO_STRING,
+	optimizationBailout: OFF_FOR_TO_STRING,
+	children: NORMAL_ON,
+	source: NORMAL_OFF,
+	moduleTrace: NORMAL_ON,
+	errors: NORMAL_ON,
+	errorDetails: OFF_FOR_TO_STRING,
+	warnings: NORMAL_ON,
+	publicPath: OFF_FOR_TO_STRING,
+	excludeModules: () => [],
+	excludeAssets: () => [],
+	maxModules: (o, { forToString }) => (forToString ? 15 : Infinity),
+	modulesSort: () => "id",
+	chunksSort: () => "id",
+	assetsSort: () => "name",
+	outputPath: OFF_FOR_TO_STRING,
+	colors: () => false
+};
+
+const normalizeExclude = item => {
+	if (typeof item === "string") {
+		const regExp = new RegExp(
+			`[\\\\/]${item.replace(
+				// eslint-disable-next-line no-useless-escape
+				/[-[\]{}()*+?.\\^$|]/g,
+				"\\$&"
+			)}([\\\\/]|$|!|\\?)`
+		);
+		return ident => regExp.test(ident);
+	}
+	if (item && typeof item === "object" && typeof item.test === "function") {
+		return ident => item.test(ident);
+	}
+	if (typeof item === "function") {
+		return item;
+	}
+	if (typeof item === "boolean") {
+		return () => item;
+	}
+};
+
+const NORMALIZER = {
+	excludeModules: value => {
+		if (!Array.isArray(value)) {
+			value = value ? [value] : [];
+		}
+		return value.map(normalizeExclude);
+	},
+	excludeAssets: value => {
+		if (!Array.isArray(value)) {
+			value = value ? [value] : [];
+		}
+		return value.map(normalizeExclude);
+	},
+	warningsFilter: value => {
+		if (!Array.isArray(value)) {
+			value = value ? [value] : [];
+		}
+		return value.map(filter => {
+			if (typeof filter === "string") {
+				return (warning, warningString) => warningString.includes(filter);
+			}
+			if (filter instanceof RegExp) {
+				return (warning, warningString) => filter.test(warningString);
+			}
+			if (typeof filter === "function") {
+				return filter;
+			}
+			throw new Error(
+				`Can only filter warnings with Strings or RegExps. (Given: ${filter})`
+			);
+		});
+	}
+};
+
+class DefaultStatsPresetPlugin {
+	/**
+	 * @param {Compiler} compiler webpack compiler
+	 * @returns {void}
+	 */
+	apply(compiler) {
+		compiler.hooks.compilation.tap("DefaultStatsPresetPlugin", compilation => {
+			compilation.hooks.statsPreset
+				.for(false)
```

Comment:
This is not compatible with `tapable` definitions. They must be updated in order to accept non-string keys.